src/field.rs

```diff
@@ -76,26 +101,41 @@ pub trait FieldElement:
     /// Returns the prime modulus `p`.
     fn modulus() -> Self::Integer;
 
-    /// Writes the field element to the end of input buffer. Exactly `BYTES` bytes will be written.
+    /// Interprets the next `Self::ENCODING.size` bytes from the input slice as an element of the
+    /// field. The bytes are interpreted according to `Self::ENCODING.order`. The `m` most
+    /// significant bits are cleared, where `m` is equal to the length of `Integer` in bits minus
+    /// the length of the modulus in bits.
     ///
-    /// TODO(acmiyaguchi) Replace this with an implementation of the corresponding serde trait
-    fn append_to(&self, bytes: &mut Vec<u8>);
-
-    /// Interprets the next `BYTES` bytes from the input buffer as an element of the field. An
-    /// error is returned if the bytes encode an integer larger than the field modulus.
+    /// # Errors
     ///
-    /// TODO(acmiyaguchi) Replace this with an implementation of the corresponding serde trait
-    fn read_from(bytes: &[u8]) -> Result<Self, FieldError>;
-
-    /// Interprets the next `BYTES` bytes from the input buffer as an element of the field. The `m`
-    /// most significant bits are cleared, where `m` is equal to the length of `Integer` in bits
-    /// minus the length of the modulus in bits. An error is returned if the result encodes an
-    /// integer larger than the field modulus.
+    /// An error is returned if the provided slice is too small to encode a field element or if the
+    /// result encodes an integer larger than the field modulus.
+    ///
+    /// # Warnings
     ///
-    /// WARNING: This function is used to convert a random byte string into a field element. It
-    /// *should not* be used to deserialize field elements.
+    /// This function is used to convert a random byte string into a field element. Use
+    /// `try_from_reader` to deserialize field elements.
     fn try_from_random(bytes: &[u8]) -> Result<Self, FieldError>;
 
+    /// Interprets the next `Self::ENCODING.size` bytes from the input slice as an element of the
+    /// field. The bytes are interpreted according to `Self::ENCODING.order`.
+    ///
+    /// # Errors
+    ///
+    /// An error is returned if the provided slice is too small to encode a field element or if the
+    /// result encodes an integer larger than the field modulus.
+    ///
+    /// # Warnings
+    ///
+    /// This function is not suitable for converting input that is not already a canonicalized
+    /// representation of a field element. Use `try_from_random` to convert non-canonical
+    /// representations.
```

Comment:
Good point -- rather than heeding a warning in documentation, callers should feel free to send any old garbage into `try_from_reader` and be ready to deal with the errors it returns.

I also get your point about making it more obvious that `try_from_random` is different from `try_from_reader`, but I'm not sure how to do that. Changing it to `try_from_prng(prng::Prng)` would be awkward but would make it more clear that the two aren't interchangeable?